### PR TITLE
Notification Plugin docs

### PR DESCRIPTION
**Pull request by dangeross**
This PR adds detailed documents on how to integrate the Notification Plugin into both iOS and Android applications.

- [x] iOS
- [x] Android

### CODE REVIEW

File: src/notifications/custom_notifications.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+<section>
+
+```kotlin,ignore
+```
+
+</section>
+</custom-tabs>
+
+
+You can customise the Notification Plugin even further by handling your own notification types sent to the application via push notification. Follow the code example below for adding these to iOS and Android.
```
Comment (dangeross) [inline src/notifications/custom_notifications.md:43]: Should be a description of the task/job implementation

**Review by roeierez — APPROVED** [review]
LGTM. This seems to close all the gaps developers may have!

File: src/notifications/getting_started.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Introduction
+
+The Breez SDK Notification Plugin provides vendors the ability to receive events via mobile notifications even while the application is in the background or closed. This plugin processes several different notification types like receiving payments, LNURL-pay flows and swap confirmations. Within your application you can even extend the functionality to handle your own notification types.
```
Comment (erdemyerebasmaz) [inline src/notifications/getting_started.md:3]:
```suggestion
The Breez SDK Notification Plugin provides vendors the ability to receive events via mobile notifications even while the application is in the background or closed. This plugin processes several different notification types like receiving payments, LNURL-pay flows and swap confirmations. You can even extend the functionality to handle your own notification types within your application.
```

File: src/notifications/ios_plugin.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
+# Add the iOS Notification Plugin
```
Comment (erdemyerebasmaz) [inline src/notifications/ios_plugin.md:1]: Perhaps we can link [breez-sdk-swift README](https://github.com/breez/breez-sdk-swift/blob/main/README.md) for other installation methods.

File: src/notifications/setup_nds.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Setup a Notification Delivery Service (NDS)
+
+Receiving push notifications involves using a NDS as an intermediary to receive the webhook event from one of the SDK services. These can be currently one of several services that provide information about events that the Breez SDK registers for. For example, payment events from the LSP or swap transaction confirmation events from the chain service. The NDS then processes this information and dispatches a push notification to the intended mobile device, ensuring the user receives timely updates about incoming events. This architecture necessitates vendors setup up and maintain their own NDS, tailored to handle and forward these notifications efficiently. An example payload when a `payment_received` POST request to the webhook URL contains the following JSON formatted structure:
```
Comment (erdemyerebasmaz) [inline src/notifications/setup_nds.md:3]:
```suggestion
Receiving push notifications involves using an NDS as an intermediary to receive the webhook event from one of the SDK services. These can be currently one of several services that provide information about events that the Breez SDK registers for. For example, payment events from the LSP or swap transaction confirmation events from the chain service. The NDS then processes this information and dispatches a push notification to the intended mobile device, ensuring the user receives timely updates about incoming events. This architecture necessitates vendors set up and maintain their own NDS, tailored to handle and forward these notifications efficiently. An example payload when a `payment_received` POST request to the webhook URL contains the following JSON formatted structure:
```

File: src/notifications/setup_nds.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+```
+{
+    "template": "payment_received",
+    "data": {  
+        "payment_hash": [payment hash]
+    }
+}
+```
+
+The vendor needs to run their own NDS because it is configured to send push notifications to your application users and therefore should be configured with the required keys and certificates. You can use our [reference NDS implementation](https://github.com/breez/notify) as an implementation starting point or as is. Our implementation of the NDS expects URLs in the following format:
```
Comment (erdemyerebasmaz) [inline src/notifications/setup_nds.md:14]:
```suggestion
The vendor needs to run their own NDS because it is configured to send push notifications to your application users and therefore should be configured with the required keys and certificates. You can use our [reference NDS implementation](https://github.com/breez/notify) as a starting point or as is. Our implementation of the NDS expects URLs in the following format:
```

File: src/notifications/register_webhook.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
+# Register a webhook
+
+Once your vendor [NDS is setup](setup_nds.md) and can accept POST requests from the SDK services, you can within your mail application register the webhook URL with the Breez SDK by calling the register webhook API as follows:
+
+<custom-tabs category="lang">
+<div slot="title">Rust</div>
+<section>
+
+```rust,ignore
+{{#include ../../snippets/rust/src/webhook.rs:register-payment-webook}}
```
Comment (erdemyerebasmaz) [inline src/notifications/register_webhook.md:10]: Anchor name "register-payment-webook" should be corrected on another PR.

File: src/notifications/register_webhook.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
+# Register a webhook
+
+Once your vendor [NDS is setup](setup_nds.md) and can accept POST requests from the SDK services, you can within your mail application register the webhook URL with the Breez SDK by calling the register webhook API as follows:
```
Comment (erdemyerebasmaz) [inline src/notifications/register_webhook.md:3]:
```suggestion
Once your vendor [NDS is set up](setup_nds.md) and can accept POST requests from the SDK services, you can within your mail application register the webhook URL with the Breez SDK by calling the register webhook API as follows:
```

File: src/notifications/register_webhook.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+<div slot="title">C#</div>
+<section>
+
+```cs,ignore
+{{#include ../../snippets/csharp/Webhook.cs:register-payment-webook}}
+```
+</section>
+</custom-tabs>
+
+Now when the NDS receives a POST request for this webhook URL, it will forward the request data via push notification to the applications Service Extension (iOS) or Foreground Service (Android) to be handled by the [Notification Plugin](setup_plugin.md).
```
Comment (erdemyerebasmaz) [inline src/notifications/register_webhook.md:71]:
```suggestion
When the NDS receives a POST request for the registered webhook URL, it will forward the request data via push notification to the applications Service Extension (iOS) or Foreground Service (Android) to be handled by the [Notification Plugin](setup_plugin.md).
```

File: src/notifications/setup_plugin.md [hunk 9]
```diff
@@ -0,0 +1,7 @@
+# Notification Integration
+
+The Notification Plugin is designed to be used when the application isn't running. For this to be acheived reliably and to ensure that the notification is processed, specific implementations are required for both Android and iOS platforms to handle the incoming push notifications.
```
Comment (erdemyerebasmaz) [inline src/notifications/setup_plugin.md:3]:
```suggestion
The Notification Plugin is designed to be used when the application isn't running. For this to be achieved reliably and to ensure that the notification is processed, specific implementations are required for both Android and iOS platforms to handle the incoming push notifications.
```

File: src/notifications/android_plugin.md [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+
+dependencies {
+    // Add the breez-sdk dependency
+    implementation "com.github.breez:breez-sdk"
+}
+```
+
+## Integrate the Notification Plugin
+
+You're ready to add some Kotlin code to implement the Notification Plugin in your application. In the example below we are using the `FirebaseMessagingService` to receive the message intents. First lets implement the Notification Plugin's `MessagingService` class along with `FirebaseMessagingService`.
```
Comment (erdemyerebasmaz) [inline src/notifications/android_plugin.md:31]:
```suggestion
You're ready to add some Kotlin code to implement the Notification Plugin in your application. In the example below, we are using the `FirebaseMessagingService` to receive the message intents. First, let's implement the Notification Plugin's `MessagingService` class along with `FirebaseMessagingService`. 
```

**Review by erdemyerebasmaz — APPROVED** [review]
Looking great! Covers pretty much every thing.
